injectors/anvil-injector/src/test/kotlin/io/mcarle/konvert/injector/koin/AnvilInjectorITest.kt

```diff
@@ -0,0 +1,382 @@
+        val (compilation) = super.compileWith(
+            listOf(SameTypeConverter()),
+            otherConverters = emptyList(),
+            expectSuccess = true,
+            options = mapOf(
+                KONVERTER_GENERATE_CLASS to "true",
+                DEFAULT_INJECTION_METHOD to "factory",
+                DEFAULT_SCOPE to "test.module.AppScope"
+            ),
+            SourceFile.kotlin(
+                name = "TestCode.kt",
+                contents =
+                """
+package test.module
+
+import io.mcarle.konvert.api.Konverter
+import io.mcarle.konvert.api.Konvert
+
+abstract class AppScope private constructor()
+
+@Konverter
+interface Mapper {
+    @Konvert
+    fun toTarget(source: SourceClass): TargetClass
+}
+
+class SourceClass(val property: String)
+class TargetClass(val property: String)
+                """.trimIndent()
+            )
+        )
+        val mapperCode = compilation.generatedSourceFor("MapperKonverter.kt")
+        println(mapperCode)
+
+        assertContains(mapperCode, "import javax.inject.Inject")
+        assertContains(mapperCode, "@Inject")
+        assertContains(mapperCode, "public constructor()")
+
+        assertContains(mapperCode, "com.squareup.anvil.annotations.ContributesBinding")
+        assertContains(mapperCode, "@ContributesBinding")
+        assertContains(mapperCode, "scope = AppScope::class")
+    }
+
+    @Test
+    fun contributesBindingAsSingletonByDefault() {
+        val (compilation) = super.compileWith(
+            listOf(SameTypeConverter()),
+            otherConverters = emptyList(),
+            expectSuccess = true,
+            options = mapOf(
+                KONVERTER_GENERATE_CLASS to "true",
+                DEFAULT_INJECTION_METHOD to "singleton",
+                DEFAULT_SCOPE to "test.module.AppScope"
+            ),
+            SourceFile.kotlin(
+                name = "TestCode.kt",
+                contents =
+                """
+package test.module
+
+import io.mcarle.konvert.api.Konverter
+import io.mcarle.konvert.api.Konvert
+
+abstract class AppScope private constructor()
+
+@Konverter
+interface Mapper {
+    @Konvert
+    fun toTarget(source: SourceClass): TargetClass
+}
+
+class SourceClass(val property: String)
+class TargetClass(val property: String)
+                """.trimIndent()
+            )
+        )
+        val mapperCode = compilation.generatedSourceFor("MapperKonverter.kt")
+        println(mapperCode)
+
+        assertContains(mapperCode, "import javax.inject.Inject")
+        assertContains(mapperCode, "@Inject")
+        assertContains(mapperCode, "public constructor()")
+
+        assertContains(mapperCode, "com.squareup.anvil.annotations.ContributesBinding")
+        assertContains(mapperCode, "@ContributesBinding")
+        assertContains(mapperCode, "scope = AppScope::class")
+
+        assertContains(mapperCode, "javax.inject.Singleton")
+        assertContains(mapperCode, "@Singleton")
```

Comment:
instead of using `assertContains` in all tests, maybe (at least once) check the whole generated code e.g. using `assertSourceEquals`. The tests will be much more readable then I suppose :)

Reply:
It is in fact more readable, but for some reason intelliJ is shouting to me about warnings in the kotlin code within `"""` brackets. I don't see a way to disable inspection for inline code, do you know if there is any way to do that?

Reply:
You could add `@Suppress` for e.g. the single statement:

![image](https://user-images.githubusercontent.com/32078713/236564601-1ec847cf-da6a-4519-a650-7413bc3bcbb5.png)

Reply:
Yea, but that's only for given code block. I'm looking for an option to disable the analysis altogether for all kotlin code blockes inside triple quotes since that just seems silly in case of testing generated code :D